Add section on migrating applications to MCS (#1251)
submariner-io/submariner-website: src/content/operations/usage/_index.en.md

$ kubectl delete namespace nginx-test
namespace "nginx-test" deleted
```

### Migrating existing applications to use MultiCluster Services

> To use MultiCluster Services applications need to use `clusterset.local` suffix. There are two main ways to achieve this
> without needing to rewrite the applications.

Reviewer: ```suggestion
To use MultiCluster Services, applications need to use the `clusterset.local` suffix.
There are two main ways to achieve this without needing to rewrite the applications.
```

#### Use DnsConfig

> First option is to use [dnsConfig][dnsconfig doc] field in the `Deployment` or `Pod` spec of the applications to add `clusterset.local`

Reviewer: ```suggestion
The first option is to use the [dnsConfig][dnsconfig doc] field in the `Deployment` or `Pod` spec of the applications to add `clusterset.local`
```

to the search path in applications' `/etc/hosts/` file.

```bash
apiVersion: v1
kind: Pod
metadata:
  namespace: default
  name: dns-example
spec:
  containers:
    - name: test
      image: nginx
  dnsPolicy: "None"
  dnsConfig:
    nameservers:
      - 192.0.2.1 # this is an example
    searches:
      - default.svc.cluserset.local
      - ns1.svc.cluserset.local # example namespace
      - svc.clusterset.local
      - clusterset.local
```

##### Pros of DnsConfig

* Useful when specific applications want to use multiple MultiCluster Services.

> * Preferred if some applications want to continue using local version of the Service

Reviewer: ```suggestion
* Preferred if some applications want to continue using the local version of the Service explicitly
```

##### Cons of DnsConfig

> *Need to modify `Deployment` or `Pod` spec of all applications that need to use MultiCluster Service.

Reviewer: ```suggestion
* It is necessary to modify the `Deployment` or `Pod` spec of all applications that need to use MultiCluster Services.
```

#### Use ExternalName Service

> Second option is to create a Service of type `ExternalName` that points to MultiClusterService.

Reviewer: ```suggestion
The second option is to create a Service of type `ExternalName` that points to a MultiClusterService.
```

```bash
apiVersion: v1
kind: Service
metadata:
  name: mcs-sample
  namespace: demo
spec:
  type: ExternalName
  sessionAffinity: None
  externalName: mcs-sample.demo.svc.clusterset.local
```

> This will make DNS replace any queries to local service `mcs-sample.demo.cluster.local` with the MultiCluster Service
> `mcs-sample.demo.svc.clusterset.local`. This can be also used to redirect to a MultiCluster Service in a different
> namespace.

Reviewer: ```suggestion
This will make DNS replace any queries to the local `mcs-sample.demo.cluster.local` service with the MultiCluster
`mcs-sample.demo.svc.clusterset.local` service.
This can be also used to redirect to a MultiCluster Service in a different namespace.
```

##### Pros of ExternalName

> * Useful all applications need to use a specific MultiCluster Service

Reviewer: ```suggestion
* Useful if all applications need to use a specific MultiCluster Service
```

* No changes to `Deployment` or `Pod` spec

##### Cons of ExternalName

> * Can't have a local version of the service as the `ExternalName` type Service will replace the existing local

Reviewer: ```suggestion
* Such services can’t be deployed locally since the `ExternalName` type Service will replace the existing local
```

> ClusterIP Service. This can be mitigated by using a different namesapce for the MultiCluster Service.

Reviewer: ```suggestion
ClusterIP Service. This can be mitigated by using a different namespace for the MultiCluster Service.
```

To use multi-cluster Services, applications need to use the `clusterset.local` suffix.

There are two main ways to achieve this without needing to rewrite the applications.

#### Use DnsConfig

The first option is to use the [dnsConfig][dnsconfig doc] field in the `Deployment` or `Pod` spec of the applications
to add `clusterset.local` to the search path in applications' `/etc/hosts/` file.

```bash
apiVersion: v1
kind: Pod
metadata:
  namespace: default
  name: dns-example
spec:
  containers:
    - name: test
      image: nginx
  dnsPolicy: "None"
  dnsConfig:
    nameservers:
      - 192.0.2.1 # this is an example
    searches:
      - default.svc.cluserset.local
      - ns1.svc.cluserset.local # example namespace
      - svc.clusterset.local
      - clusterset.local
```

##### Pros of DnsConfig

* Useful when specific applications want to use multiple multi-cluster Services.
* Preferred if some applications want to continue using the local version of the Service explicitly.

##### Cons of DnsConfig

* It is necessary to modify the `Deployment` or `Pod` spec of all applications that need to use multi-cluster Services.

#### Use ExternalName Service

> The second option is to create a Service of type `ExternalName` that points to multi-clusterService.

Reviewer: ```suggestion
The second option is to create a Service of type `ExternalName` that points to a multi-cluster Service.
```
